tasks/apply-mapping/apply-mapping.yaml

```diff
@@ -100,12 +115,24 @@ spec:
                   var_name="${BASH_REMATCH[1]}"
 
                   # Sanity check of the template variable name
-                  if [[ ! "$var_name" =~ ^[a-zA-Z0-9._-]+$ ]]; then
+                  if [[ "$var_name" != "incrementer" && ! "$var_name" =~ ^[a-zA-Z0-9._-]+$ ]]; then
                     echo "Error: Invalid variable name in tag definition: $var_name" >&2
                     exit 1
                   fi
 
-                  replacement=$(substitute "$var_name" "$substitute_map" "$labels_map")
+                  # Handle incrementer logic
+                  if [[ "$var_name" == "incrementer" ]]; then
+                      repo=$(jq -r '.repository' <<< "$component")
+                      tag_prefix=$(echo "$tag" | sed 's/{{ *incrementer *}}//')
```

Comment:
Won't this remove `{{ *incrementer *}}` from the variable? What even is that? The `*` has no space before the word but does after

Reply:
Never heard of regular expressions? ;) ` *` means zero or more spaces. And yeah, this would remove it from the tag string.

Reply:
The variable is called `tag_prefix`, so my guess is that Happy was expecting the incrementer to always be at the end. So he would remove it and then add the correct value. But as you pointed out, it doesn't have to be at the end.

> Will the regex expand in single quotes?

We don't want or need bash to do anything about the regexp before it passes it to `sed`. Similar to when you do something like `jq '.something | length'` - you typically use single quotes, because you don't want bash to process that string in any way.